specification/marketplacecatalog/resource-manager/Products/preview/2022-09-01-preview/products.json

```diff
@@ -0,0 +1,1411 @@
+{
+  "x-generator": "NSwag v13.12.1.0 (NJsonSchema v10.4.6.0 (Newtonsoft.Json v13.0.0.0))",
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Marketplace Catalog Browse API",
+    "description": "Microsoft Marketplace Catalog Browse API",
+    "version": "2022-09-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Marketplace/products/{id}": {
+      "get": {
+        "tags": [
+          "SinglePrivateProduct"
+        ],
+        "summary": "Returns a single product directly from BigCat, by subscription as audience",
+        "operationId": "SinglePrivateProduct_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "type": "string",
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "description": "The unique product Id",
+            "x-nullable": true
```

Comment:
>"x-nullable": true [](http://example.com/codeflow?start=12&length=18)

You want to allow "null" in product id, not sure how will it translate in whole URI path?

Reply:
Have removed references to x-nullable from the document